numpy/_core/src/umath/ufunc_object.c

```diff
@@ -2889,15 +2896,14 @@ PyUFunc_Accumulate(PyUFuncObject *ufunc, PyArrayObject *arr, PyArrayObject *out,
          * Output (dataptr[0]) and input (dataptr[1]) may point to the
          * same memory, e.g. np.add.accumulate(a, out=a).
          */
-        if (descrs[2]->type_num == NPY_OBJECT) {
-            /*
-             * Incref before decref to avoid the possibility of the
-             * reference count being zero temporarily.
-             */
-            Py_XINCREF(*(PyObject **)dataptr_copy[1]);
-            Py_XDECREF(*(PyObject **)dataptr_copy[0]);
-            *(PyObject **)dataptr_copy[0] =
-                                *(PyObject **)dataptr_copy[1];
+        if (copy_info.func) {
+            const npy_intp one = 1;
+            const npy_intp strides[2] = {itemsize, itemsize};
+            if (copy_info.func(
+                    &copy_info.context, &dataptr_copy[1], &one,
+                    strides, copy_info.auxdata) < 0) {
+                goto fail;
+            }
         }
         else {
             memmove(dataptr_copy[0], dataptr_copy[1], itemsize);
```

Comment:
```suggestion
            memmove(dataptr_copy[2], dataptr_copy[1], itemsize);
```

Just to make it match the other branch you changed here. I was confused last Friday but then I see now that 2 and 0 are the same pointer.

Reply:
Ah, thanks, missed that.  Better to make it uniform!